sotopia/ui/README.md

```diff
@@ -0,0 +1,198 @@
+
+Update agent profile in the API server.
+Request Body:
+AgentProfile
+
+returns:
+- agent_id: str
+
+
+#### PUT /scenarios/{scenario_id}
+
+Update scenario profile in the API server.
+Request Body:
+EnvironmentProfile
+
+returns:
+- scenario_id: str
+
+### Initiating a new non-streaming simulation episode
+
+#### POST /episodes/
+
+```python
+class SimulationEpisodeInitiation(BaseModel):
+    scenario_id: str
+    agent_ids: list[str]
+    episode_tag: str
+    models: list[str]
+```
+
+Send episode profile to the API server.
+Request Body:
+SimulationEpisodeInitiation
+
+returns:
+- episode_id: str (This is the id of the episode that will be used to get the episode data, saved in the redis database)
+
+### Initiating a new interactive streaming simulation episode (this operation will open a websocket connection)
+
+We use the websocket connection to send the simulation step-by-step results to the UI.
+Please see an example protocol [here](https://claude.site/artifacts/322011f6-597f-4819-8afb-bf8137dfb56a)
+
+#### WEBSOCKET /ws/simulate/?token={token}
+
+Parameters:
+- Token: String. User authentication token. Each token maps to a unique session.
+
+returns:
+- msg: WSMessage
+
+**WSMessage**
+```json
+{
+    "type": "WSMessageType",
+    "data": {
+        // Message-specific payload
+    }
+}
+```
+
+**WSMessageType**
+| Type | Direction   | Description |
+|-----------|--------|-------------|
+| SERVER_MSG | Server → Client | Standard message from server (payload: `messageForRendering` [here](https://github.com/sotopia-lab/sotopia-demo/blob/main/socialstream/rendering_utils.py) ) |
+| CLIENT_MSG | Client → Server | Standard message from client (payload: TBD) |
+| ERROR      | Server → Client | Error notification (payload: TBD) |
+| START_SIM  | Client → Server | Initialize simulation (payload: `SimulationEpisodeInitialization`) |
+| END_SIM    | Client → Server | End simulation (payload: not needed) |
+| FINISH_SIM | Server → Client | Terminate simulation (payload: not needed) |
+
+
+**Error Type**
+
+| Error Code | Description |
+|------------|-------------|
+| NOT_AUTHORIZED | Authentication failure - invalid or expired token |
+| SIMULATION_ALREADY_STARTED | Attempt to start a simulation that is already active |
+| SIMULATION_NOT_STARTED | Operation attempted on an inactive simulation |
+| RESOURCE_NOT_FOUND | Required env_id or agent_ids not found |
+| SIMULATION_ERROR |  Error occurred during simulation execution |
+| SIMULATION_INTERRUPTED | The simulation is interruped |
+| OTHER | Other unspecified errors |
+
+
+
+**Implementation plan**: Currently only support LLM-LLM simulation based on [this function](https://github.com/sotopia-lab/sotopia/blob/19d39e068c3bca9246fc366e5759414f62284f93/sotopia/server.py#L108).
+
+
+## An example to run simulation with the API
+
+**Get all scenarios**:
+```bash
+curl -X GET "http://localhost:8000/scenarios"
+```
+
+Randomly select a scenario, e.g., `01HZRGTG1K4YQ2CBS9SNH28R9S`
```

Comment:
#### GET /scenarios

According to the doc:

```
Get all scenarios.

returns:
- scenarios: list[EnvironmentProfile]
```

It should not be a random scenario?

Reply:
typo